src/components/ZetkinCalendar/index.tsx

```diff
@@ -0,0 +1,114 @@
+            setFocusDate(new Date(focusDate.getFullYear(), focusDate.getMonth() + 1, focusDate.getDate()));
+        }
+        else if (range === CALENDAR_RANGES.WEEK) {
+            setFocusDate(new Date(new Date(focusDate).setDate(focusDate.getDate() + 7)));
+        }
+    };
+
+    const handleBackButtonClick = () => {
+        if (range === CALENDAR_RANGES.MONTH) {
+            setFocusDate(new Date(focusDate.getFullYear(), focusDate.getMonth() - 1, focusDate.getDate()));
+        }
+        else if (range === CALENDAR_RANGES.WEEK) {
+            setFocusDate(new Date(new Date(focusDate).setDate(focusDate.getDate() - 7)));
+        }
+    };
+
+    return (
+        <Box display="flex" flexDirection="column" height={ 1 }>
+            <Box display="grid" flexGrow={ 0 } gridTemplateAreas={ `". nav view"` } gridTemplateColumns="repeat(3, minmax(0, 1fr))">
+                <Box alignItems="center" display="flex" gridArea="nav" justifyContent="center">
+                    <Button color="primary" data-testid="back-button" onClick={ handleBackButtonClick }>
+                        <Msg id="misc.calendar.prev" />
+                    </Button>
+                    <Box data-testid="selected-month" p={ 1 } textAlign="center" width="10rem">
+                        <Typography>
+                            { range === CALENDAR_RANGES.MONTH && (<FormattedDate
+                                month="long"
+                                value={ new Date(focusDate.getUTCFullYear(), focusDate.getUTCMonth() + 1, 0) }
+                                year="numeric"
+                            />) }
+                            { range === CALENDAR_RANGES.WEEK && (new Date(new Date(
+                                new Date(focusDate).setDate(
+                                    new Date(focusDate).getDate() - new Date(focusDate).getDay() + 1,
+                                ),
+                            ).setHours(0, 0, 0, 0))).getWeekNumber() }
+                        </Typography>
+                    </Box>
+                    <Button color="primary" data-testid="fwd-button" onClick={ handleForwardButtonClick }>
+                        <Msg id="misc.calendar.next" />
+                    </Button>
+                </Box>
+                <Box alignItems="center" display="flex" gridArea="view" justifySelf="end" mr={ 1 }>
+                    <TextField
+                        aria-label={ intl.formatMessage(
+                            { id: 'misc.calendar.label' }) }
+                        id="calendar-view"
+                        onChange={ e => setRange(e.target.value as CALENDAR_RANGES) }
+                        select
+                        value={ range }>
+                        { calendars.map(item => (
+                            <MenuItem key={ item.id } value={ item.id }>
+                                { item.name }
+                            </MenuItem>
+                        )) }
```

Comment:
Instead of manually making the `calendars` variable above, you could just use the `CALENDAR_RANGES` enum to automatically generate the range options.

```suggestion
                        { Object.values(CALENDAR_RANGES).map(range => (
                            <MenuItem key={ range } value={ range }>
                                <Msg id={`misc.calendar.${range}`} />
                            </MenuItem>
                        )) }
```

Reply:
genius

Reply:
Thank you :) It's not obvious enums are just objects at runtime.